edk2toollib/windows/telem/cper_parser_tool.py

```diff
@@ -0,0 +1,659 @@
+
+    def FlagsParse(self) -> str:
+        '''Check the flags field and populate list containing applicable flags'''
+
+        FlagList = []
+
+        # Check each bit for associated flag
+        if(self.flags & int('0b1', 2)):
+            FlagList.append("Recovered")
+        if(self.flags & int('0b10', 2)):
+            FlagList.append("Previous Error")
+        if(self.flags & int('0b100', 2)):
+            FlagList.append("Simulated")
+        if(self.flags & int('0b1000', 2)):
+            FlagList.append("Device Driver")
+        if(self.flags & int('0b10000', 2)):
+            FlagList.append("Critical")
+        if(self.flags & int('0b100000', 2)):
+            FlagList.append("Persist")
+
+        # If no flags were found
+        if(FlagList == []):
+            return "None"
+
+        # Join the FlagsList elements separated by commas
+        return ", ".join(FlagList)
+
+    def PersistenceInfoParse(self) -> str:
+        '''Parse the persistence info which is produced and consumed by the creator of the Error Record'''
+        return str(self.persistenceInfo)
+
+    def PrettyPrint(self) -> str:
+        '''Print relevant portions of the CPER header. Change to suit your needs'''
+
+        string = ""
+        temp = ""
+
+        string += "Record Length:  " + str(self.recordLength) + '\n'
+
+        temp = self.TimestampParse()
+        string += "Time of error:  " + temp + '\n'
+
+        temp = self.ErrorSeverityParse()
+        string += "Error severity: " + temp + '\n'
+
+        temp = self.FlagsParse()
+        string += "Flags Id:       " + temp + '\n'
+
+        temp = self.PlatformIdParse()
+        string += "Platform Id:    " + temp + '\n'
+
+        temp = self.PartitionIdParse()
+        string += "Partition Id:   " + temp + '\n'
+
+        temp = self.CreatorIdParse()
+        string += "Creator Id:     " + temp + '\n'
+
+        return string[0:-1]  # omit the last newline
+
+
+class CPER_SECTION_HEADER(object):
+    '''TODO: Fill in'''
+
+    STRUCT_FORMAT = "=IIHccI16s16sI20s"
+
+    def __init__(self, input: str):
+
+        self.FlagList = []  # go to self.FlagsParse() to see description of this field
+        self.fruIdValid = False
+        self.fruStringValid = False
+
+        (self.sectionOffset,
+            self.sectionLength,
+            self.revision,
+            self.validationBits,
+            self.reserved,
+            self.flags,
+            self.sectionType,
+            self.fruId,
+            self.sectionSeverity,
+            self.fruString) = struct.unpack_from(self.STRUCT_FORMAT, input)
+
+        self.FlagsParse()
+        self.ValidBitsParse()
+
+    def RevisionParse(self) -> str:
+        '''
+        Parse the major and minor version number of the Error Record definition
+
+        TODO: Actually parse the zeroth and first byte which represent the minor
+        and major version numbers respectively
+        '''
+
+        return str(self.revision)
```

Comment:
You do RevisionParsing quite a bit. Is there any commonality between these different classes?